extensions-twitter/scribe-emitter/src/main/java/org/apache/druid/emitter/scribe/ScribeAdminLogEntry.java

```diff
@@ -0,0 +1,112 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.emitter.scribe;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.druid.java.util.emitter.core.Event;
+import org.apache.druid.java.util.emitter.service.ServiceMetricEvent;
+import org.apache.thrift.TBase;
+
+public class ScribeAdminLogEntry
+{
+  private static final String AUDIT_KEY = "key";
+  private static final String AUDIT_TYPE = "type";
+  private static final String AUTHOR = "author";
+  private static final String COMMENT = "comment";
+  private static final String IP = "remote_address";
+  private static final String CREATED_DATE = "created_date";
+  private static final String PAYLOAD = "payload";
+
+  private String audit_key;
+  private String audit_type;
+  private String author;
+  private String comment;
+  private String remote_address;
+  private String created_date;
+  private String payload;
+
+  private String role;
+  private String druid_version;
+  private String environment;
+  private String dataCenter;
+  private String cluster_name;
+
+  public ScribeAdminLogEntry(Event event, ScribeEmitterConfig config, ObjectMapper jsonMapper) throws JsonProcessingException
+  {
+    ServiceMetricEvent adminLogEvent = (ServiceMetricEvent) event;
+
+    this.audit_key = (String) adminLogEvent.getUserDims().getOrDefault(AUDIT_KEY, null);
+    this.audit_type = (String) adminLogEvent.getUserDims().getOrDefault(AUDIT_TYPE, null);
+    this.author = (String) adminLogEvent.getUserDims().getOrDefault(AUTHOR, null);
+    this.comment = (String) adminLogEvent.getUserDims().getOrDefault(COMMENT, null);
+    this.remote_address = (String) adminLogEvent.getUserDims().getOrDefault(IP, null);
+    this.created_date = (String) adminLogEvent.getUserDims().getOrDefault(CREATED_DATE, null);
+    this.payload = jsonMapper.writeValueAsString(adminLogEvent.getUserDims().getOrDefault(PAYLOAD, null));
+
+    this.role = config.getRole();
+    this.druid_version = config.getDruidVersion();
+    this.environment = config.getEnvironment();
+    this.dataCenter = config.getDataCenter();
+    this.cluster_name = config.getClusterName();
+  }
+
+  public TBase toThrift()
+  {
+    DruidAdminLogEvent thriftEvent = new DruidAdminLogEvent();
+    thriftEvent.setAudit_key(audit_key);
+    thriftEvent.setAudit_type(audit_type);
+    thriftEvent.setAuthor(author);
+    thriftEvent.setComment(comment);
+    thriftEvent.setRemote_address(remote_address);
+    thriftEvent.setCreated_date(created_date);
+    thriftEvent.setPayload(payload);
+    thriftEvent.setRole(role);
+    thriftEvent.setDruid_version(druid_version);
+    thriftEvent.setEnvironment(environment);
+    thriftEvent.setDatacenter(dataCenter);
+    thriftEvent.setCluster_name(cluster_name);
+    return thriftEvent;
+  }
+
+  @Override
+  public String toString()
+  {
+    String resultString = "";
```

Comment:
I'd suggest have one multi-line string definition, rather than doing +=, to save memory copy
what do you think?

Reply:
Yes this makes sense

Reply:
fixed at both ScribeAdminLogEntry and ScribeRequestLogEntry